src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
+                C0 = Sse.Add(C0, V00);
+                Vector128<float> C3 = Sse.Add(C2, V01);
+                C2 = Sse.Subtract(C2, V01);
+                Vector128<float> C5 = Sse.Subtract(C4, V02);
+                C4 = Sse.Add(C4, V02);
+                Vector128<float> C7 = Sse.Add(C6, V03);
+                C6 = Sse.Subtract(C6, V03);
+
+                C0 = Sse.Shuffle(C0, C1, 0xD8);
+                C2 = Sse.Shuffle(C2, C3, 0xD8);
+                C4 = Sse.Shuffle(C4, C5, 0xD8);
+                C6 = Sse.Shuffle(C6, C7, 0xD8);
+
+                C0 = Sse.Shuffle(C0, C0, 0xD8);
+                C2 = Sse.Shuffle(C2, C2, 0xD8);
+                C4 = Sse.Shuffle(C4, C4, 0xD8);
+                C6 = Sse.Shuffle(C6, C6, 0xD8);
+
+                // Get the determinant
+                vTemp2 = row1;
+                Vector128<float> vTemp = Sse.Multiply(C0, vTemp2);
+                vTemp2 = Sse.Shuffle(vTemp2, vTemp, 0x40);
+                vTemp2 = Sse.Add(vTemp2, vTemp);
+                vTemp = Sse.Shuffle(vTemp, vTemp2, 0x30);
+                vTemp = Sse.Add(vTemp, vTemp2);
+                vTemp = Sse.Shuffle(vTemp, vTemp, 0xAA);
+
+                // Check determinant is not zero
+                if (MathF.Abs(vTemp.GetElement<float>(0)) < float.Epsilon)
+                {
+                    result = new Matrix4x4(float.NaN, float.NaN, float.NaN, float.NaN,
+                                float.NaN, float.NaN, float.NaN, float.NaN,
+                                float.NaN, float.NaN, float.NaN, float.NaN,
+                                float.NaN, float.NaN, float.NaN, float.NaN);
+                    return false;
+                }
+
+                Vector128<float> ones = Vector128.Create(1.0f, 1.0f, 1.0f, 1.0f);
+
+                vTemp = Sse.Divide(ones, vTemp);
+
+                row1 = Sse.Multiply(C0, vTemp);
+                row2 = Sse.Multiply(C2, vTemp);
+                row3 = Sse.Multiply(C4, vTemp);
+                row4 = Sse.Multiply(C6, vTemp);
+
+                result.M11 = row1.GetElement<float>(0);
+                result.M12 = row1.GetElement<float>(1);
+                result.M13 = row1.GetElement<float>(2);
+                result.M14 = row1.GetElement<float>(3);
+
+                result.M21 = row2.GetElement<float>(0);
+                result.M22 = row2.GetElement<float>(1);
+                result.M23 = row2.GetElement<float>(2);
+                result.M24 = row2.GetElement<float>(3);
+
+                result.M31 = row3.GetElement<float>(0);
+                result.M32 = row3.GetElement<float>(1);
+                result.M33 = row3.GetElement<float>(2);
+                result.M34 = row3.GetElement<float>(3);
+
+                result.M41 = row4.GetElement<float>(0);
+                result.M42 = row4.GetElement<float>(1);
+                result.M43 = row4.GetElement<float>(2);
+                result.M44 = row4.GetElement<float>(3);
+            }
```

Comment:
Is this missing a `return true` or an `else` wrapping the fallback?

Reply:
Good catch!  The other functions use the same style, but I suppose the compiler would have caught this error if the fallback was wrapped.